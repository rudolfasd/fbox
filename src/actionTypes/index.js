export * from './saga';
export * from './pointList';