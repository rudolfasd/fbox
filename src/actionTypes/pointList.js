export const PLIST_ADD_POINT = 'PLIST_ADD_POINT';
export const PLIST_DEL_POINT = 'PLIST_DEL_POINT';
export const PLIST_DEL_ALL_POINTS = 'PLIST_DEL_ALL_POINTS';
export const PLIST_REORDER_LIST = 'PLIST_REORDER_LIST';
