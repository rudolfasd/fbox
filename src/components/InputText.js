import React from 'react';


class InputText extends React.Component {

	state = {text:''};

	hookEnter = e => {
		if (e.key !== 'Enter') return ;
 		this.setState({text:''});
		if (e.target.value){
			this.props.sagaAddPoint({text: e.target.value});
		};
	};

	handleInput = e => {
		this.setState({
			text: e.target.value
		});
	};
	
	render(){
		return(
			<div className='input-text'>
				<input className='input-text-field'
					type='text' maxLength='60'
					onKeyPress={this.hookEnter.bind(this)}
					onChange={this.handleInput}
					value={this.state.text}
				/>
			</div>
		);
	};
};

export default InputText;

