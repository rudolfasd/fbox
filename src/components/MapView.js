import React from 'react';

import loadScript from '../services/loadScript';
import ymapCreate from '../services/ymap';


const MAP_PARENT_ELEMENT = 'MapView';
const scriptSrc = process.env.REACT_APP_YMAPS_SCRIPT
    .replace('<apikey>',process.env.REACT_APP_YMAPS_API_KEY);
const scriptId = 'YMaps';

export const scriptOnloadCallback = () => ymapCreate(
    MAP_PARENT_ELEMENT,
    {
        center: [57, 95],
        zoom: 3
    },{
        searchControlProvider: 'yandex#search',
        balloonPanelMaxMapArea: Infinity
    }
)


class MapView extends React.Component {

    componentDidMount(){
        loadScript(scriptSrc, scriptId, scriptOnloadCallback);
    }

    render(){
        return(
            <div className='map-view' id={MAP_PARENT_ELEMENT}></div>
        );
    }
}

export default MapView;

