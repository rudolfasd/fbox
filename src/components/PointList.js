import React from 'react';
import { SortableContainer, SortableElement } from 'react-sortable-hoc';
import arrayMove from 'array-move';


const SortableItem = SortableElement(({id,num,text,delItemFunction}) => {
	return (
		<div className='item-wrap'>
			<div className='item'>
				<div className='item-num'>{num}</div>
				<div className='item-text'>
					{text}
				</div>
				<div className='item-remove'
					onClick={() => delItemFunction({id})}>
				</div>
			</div>
		</div>
	);
});

const SortableList = SortableContainer(({order,points,delItemFunction}) => {
	return(
		<div className='point-list-overflow'>
			<div className='point-list'>
				{order.map((id, index) => (
					<SortableItem
						key={`item-${id}`}
						index={index}
						id={id}
						num={index + 1}
						text={points[id]}
						delItemFunction={delItemFunction}
					/>
				))}
			</div>
		</div>
	);
});


class PointList extends React.Component {

	order = () => this.props.pointList.order;

	onSortEnd = ({oldIndex, newIndex}) => {
		const newOrderedList = arrayMove(this.order(), oldIndex, newIndex);
		this.props.sagaReorderList({order: newOrderedList});
	};
	shouldCancelStart = (e) => {
		let classList = e.target.classList;
		if (classList.contains('item-remove')) {
			return true;
		};
		return false;
	};
	render(){
		return (
				<SortableList
					lockAxis='y'
					lockToContainerEdges={true}
					helperClass='point-dragging'
					onSortEnd={this.onSortEnd}
					shouldCancelStart={this.shouldCancelStart}
					order={this.order()}
					points={this.props.pointList.points}
					delItemFunction={this.props.sagaDelPoint}
				/>
		);
	};
};

export { SortableItem, SortableList };
export default PointList;

