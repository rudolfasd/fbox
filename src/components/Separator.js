import React from 'react';


class Separator extends React.Component {

    sepBtnClasses = () => {
        let classes = ['sep-btn-del-all'];
        let order = this.props.pointList.order;
        if (order.length === 0){
            classes.push('inactive');
        };
        return classes.join(' ');
    };
    
    delAllPoints = () => {
        let order = this.props.pointList.order;
        if (order.length){
            this.props.sagaDelAllPoints();
        };
    };

    render(){
        return(
            <div className='sep' onClick={this.delAllPoints}>
                <div className='sep-line'></div>
                <div className={this.sepBtnClasses()}></div>
            </div>
        );
    };
};

export default Separator;