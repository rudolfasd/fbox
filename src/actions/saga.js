import * as t from '../actionTypes/saga';


export function sagaAddPoint(payload){
	return{
		type: t.SAGA_ADD_POINT,
		payload
	};
};

export function sagaDelPoint(payload){
	return{
		type: t.SAGA_DEL_POINT,
		payload
	};
};

export function sagaDelAllPoints(){
	return{
		type: t.SAGA_DEL_ALL_POINTS
	};
};

export function sagaReorderList(payload){
	return{
		type: t.SAGA_REORDER_LIST,
		payload
	};
};