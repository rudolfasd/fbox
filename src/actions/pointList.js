import * as t from '../actionTypes/pointList';


export function plistAddPoint(payload){
	return{
		type: t.PLIST_ADD_POINT,
		payload
	};
};

export function plistDelPoint(payload){
	return{
		type: t.PLIST_DEL_POINT,
		payload
	};
};

export function plistDelAllPoints(){
	return{
		type: t.PLIST_DEL_ALL_POINTS
	};
};
