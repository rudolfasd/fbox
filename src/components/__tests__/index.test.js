import React from 'react';
import { Provider } from 'react-redux';

import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import  ConnectedApp, { App } from '..';
import InputText from '../InputText';
import PointList from '../PointList';
import Separator from '../Separator';
import MapView from '../MapView';

import * as a from '../../actions';
import * as t from '../../actionTypes';


jest.mock('../MapView',() => {
    return function() {
        return (<div className='map-view'></div>);
    };
});


describe('REACT COMPONENT <App />',()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper = shallow(<App />);
    });

    it('component rendered', () => {
       expect(wrapper.length).toEqual(1)
    });
    it('is a "div" with class "App">', () => {
        expect(wrapper.type()).toEqual('div');
        expect(wrapper.hasClass('App')).toEqual(true);
    });
    it('contains one "InputText"',() => {
        expect(wrapper.find(InputText).length).toEqual(1);
    });
    it('contains one "PointList"',() => {
        expect(wrapper.find(PointList).length).toEqual(1);
    });
    it('contains one "MapView"',() => {
        expect(wrapper.find(MapView).length).toEqual(1);
    });
    it('contains one "Separator"',() => {
        expect(wrapper.find(Separator).length).toEqual(1);
    });
});

describe('REACT COMPONENT <App /> with REDUX store', () => {
    const initState = { pointListState: {nextId: 2, order: [], points: {} }};
    const mockStore = configureStore();
    let store, wrapper;

    beforeEach(() => {
        store = mockStore(initState);
        wrapper = mount(<Provider store={store}><ConnectedApp /></Provider>);
    });

    it('component rendered', () => {
        expect(wrapper.find(ConnectedApp).length).toEqual(1);
    });
    it('check state-prop for pointListReducer',() => {
        expect(wrapper.find(App).prop('pointList'))
        .toEqual(initState.pointListState);
    });
});

describe('REACT-REDUX check actions dispatching',() => {
    const initState = { pointListState: {nextId: 2, order: [], points: {} }};
    const mockStore = configureStore();
    let store;

    beforeEach(() => {
        store = mockStore(initState);
    });

    it('Add point',() => {
        store.dispatch(a.plistAddPoint({text: 'Some text ...'}));
        expect(store.getActions()[0].type).toBe(t.PLIST_ADD_POINT);
    });
    it('Del point',() => {
        store.dispatch(a.plistDelPoint({id: 123}));
        expect(store.getActions()[0].type).toBe(t.PLIST_DEL_POINT);
    });
    it('Del all points',() => {
        store.dispatch(a.plistDelAllPoints());
        expect(store.getActions()[0].type).toBe(t.PLIST_DEL_ALL_POINTS);
    });
});

describe('REACT COMPONENT <App /> - Snapshot', () => {
    const initState = { pointListState: {nextId: 2, order: [], points: {} }};
    const mockStore = configureStore();
    let store, renderedValue;

    beforeEach(() => {
        store = mockStore(initState);
    });

    it('App initialization page', () => {
        renderedValue = renderer.create(
            <Provider store={store}><ConnectedApp /></Provider>
        ).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });
});
