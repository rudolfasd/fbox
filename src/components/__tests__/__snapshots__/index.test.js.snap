// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REACT COMPONENT <App /> - Snapshot App initialization page 1`] = `
<div
  className="App"
>
  <main>
    <sidebar
      is="div"
    >
      <div
        className="input-text"
      >
        <input
          className="input-text-field"
          maxLength="60"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="sep"
        onClick={[Function]}
      >
        <div
          className="sep-line"
        />
        <div
          className="sep-btn-del-all inactive"
        />
      </div>
      <div
        className="point-list-overflow"
      >
        <div
          className="point-list"
        />
      </div>
    </sidebar>
    <div
      className="map-view"
    />
  </main>
</div>
`;
